apps/web/src/stores/documentLogs.ts

```diff
@@ -12,20 +12,24 @@ export default function useDocumentLogs(
     projectId,
     page,
     pageSize,
+    onFetched,
   }: {
-    documentUuid: string
     commitUuid: string
     projectId: number
     page: string | null
     pageSize: string | null
+    documentUuid?: string
+    onFetched?: (data: DocumentLogWithMetadataAndError[]) => void
   },
   { fallbackData }: SWRConfiguration = {},
 ) {
   const fetcher = useFetcher(
-    ROUTES.api.projects
-      .detail(projectId)
-      .commits.detail(commitUuid)
-      .documents.detail(documentUuid).documentLogs.root,
+    documentUuid
```

Comment:
So the nasty hack explained in the other comment is this. 
What I do in the logs tab is to fetch a paginated list of logs of `1` page. So when user click next it goes to page 2 of size 1

The problem is that when this happens I store the `documentLogUuid` in local storage so this way next time the user visits this tab the current log is the one they navigated last.

Because I'm doing 2 fetch here
1. Fetch all logs paginated 
2. Fetch stored log by uuid 

The first fetch overrides the second one. So to avoid that I skip the fetching of logs of there is a log UUID stored in local storage. 

Any better ideas?

Reply:
i don't understand how they can override each other since they are completely different stores no? the key is different, hard to understand without actually pair programming it